src/cc/bpf_module.cc

```diff
@@ -631,17 +639,40 @@ int BPFModule::finalize() {
 
   engine_->finalizeObject();
 
-  // give functions an id
-  for (auto section : sections_)
-    if (!strncmp(FN_PREFIX.c_str(), section.first.c_str(), FN_PREFIX.size()))
-      function_names_.push_back(section.first);
-
   if (flags_ & DEBUG_SOURCE) {
-    SourceDebugger src_debugger(mod, sections_, FN_PREFIX, mod_src_,
+    SourceDebugger src_debugger(mod, *sections_p, FN_PREFIX, mod_src_,
                                 src_dbg_fmap_);
     src_debugger.dump();
   }
 
+  if (!rw_engine_enabled_) {
+    // Setup sections_ correctly and then free llvm internal memory
+    for (auto section : tmp_sections) {
+      auto fname = section.first;
+      uintptr_t size = get<1>(section.second);
+      uint8_t *tmp_p = NULL;
+      // Only copy data for non-map sections
+      if (strncmp("maps/", section.first.c_str(), 5)) {
+        uint8_t *addr = get<0>(section.second);
+        tmp_p = new uint8_t[size];
+        if (tmp_p == NULL) {
```

Comment:
and with 'new' this check becomes unnecessary.
even with malloc it's redundant. oom will not be seen by user space.

Reply:
Just checked. Understand `new` does not need checking as it will throw an exception.
Why `malloc` does not need checking? Although extremely rare, the program may run out of virtual space (e.g., user setting pretty strict RLIMIT_AS or RLIMIT_DATA)?

Reply:
In either case, if there does not exit enough physical memory and OS oom'ed, nothing we can do here. Yes, the message will be in kernel.